experimental/dds/tree2/src/feature-libraries/editable-tree-2/lazyField.ts

```diff
@@ -349,7 +349,12 @@ export class LazySequence<TTypes extends AllowedTypes>
 		const count = sourceEnd - sourceStart;
 		let destinationIndex = index;
 		if (sourceField === this) {
-			destinationIndex -= count;
+			if (destinationIndex > sourceStart) {
+				destinationIndex =
+					destinationIndex < sourceEnd
+						? sourceStart // distination overlaps with source range -> slide to left
```

Comment:
```suggestion
						? sourceStart // destination overlaps with source range -> slide to left
```
This typo made me realize that while "src" and "dest" are opposites, so are "src" and "dist".